File: Backend/Models/Medicine.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models
{
    /// <summary>
    /// Global medicine catalog (~20k rows). Platform-owned, no TenantId.
    /// Uploaded once via CSV import.
    /// </summary>
    public class Medicine
    {
        public int Id { get; set; }

        [Required]
        [StringLength(300)]
        public string Name { get; set; } = string.Empty;

        [StringLength(300)]
        public string? GenericName { get; set; }

        [StringLength(100)]
        public string? Strength { get; set; }

        [StringLength(100)]
        public string? DosageForm { get; set; }

        [StringLength(100)]
        public string Category { get; set; } = "Medicine";

        [StringLength(100)]
        public string Brand { get; set; } = "General";

        [StringLength(50)]
        public string Unit { get; set; } = "Tablet";

        [StringLength(50)]
        public string? Barcode { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;

        public ICollection<TenantMedicine> TenantMedicines { get; set; } = new List<TenantMedicine>();
    }
}

File: Frontend/POS/src/app/core/models/medicine.ts
export interface Medicine {
  id: number;
  name: string;
  genericName?: string | null;
  strength?: string | null;
  dosageForm?: string | null;
  category: string;
  brand: string;
  unit: string;
  barcode?: string | null;
  isActive: boolean;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface PagedMedicineResult {
  items: Medicine[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface MedicineImportResult {
  inserted: number;
  updated: number;
  skipped: number;
  errors: string[];
}

export const MEDICINE_TABLE_COLUMNS: { key: keyof Medicine; label: string }[] = [
  { key: 'id', label: 'ID' },
  { key: 'name', label: 'Name' },
  { key: 'genericName', label: 'Generic Name' },
  { key: 'strength', label: 'Strength' },
  { key: 'dosageForm', label: 'Dosage Form' },
  { key: 'category', label: 'Category' },
  { key: 'brand', label: 'Brand' },
  { key: 'unit', label: 'Unit' },
  { key: 'barcode', label: 'Barcode' }
];
